addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3228,6 +3228,9 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
      * @override
      */
     _computeVisibility() {
+        if (!this.originalSrc) {
+            return true; // Will display external warning
+        }
```

Comment:
Mmh should it in fact be false for unsupported mimetype or should there be a warning too? (I don't know, do what you think is best).
As discussed, you can adapt the message in JS for the t-field image optimization